src/ocrmypdf/ocr_engine/rapidocr.py

```diff
@@ -0,0 +1,440 @@
+    log.warning(f"✗ RapidOCR import failed: {IMPORT_ERROR_MESSAGE}")
+
+
+class RapidOcrEngine(OcrEngine):
+    """Implements OCR with RapidOCR."""
+
+    def __init__(self):
+        """Initialize the RapidOCR engine."""
+        if not RAPIDOCR_AVAILABLE:
+            log.warning("RapidOCR engine initialized but the library is not available")
+        else:
+            log.info("RapidOCR engine initialized successfully")
+
+    @staticmethod
+    def version() -> str:
+        """Return the version of RapidOCR."""
+        if not RAPIDOCR_AVAILABLE:
+            return "not installed"
+        return rapidocr_version
+
+    @staticmethod
+    def creator_tag(options) -> str:
+        """Return a creator tag for PDF metadata."""
+        return f"RapidOCR {RapidOcrEngine.version()}"
+
+    def __str__(self):
+        return f"RapidOCR {self.version()}"
+
+    @staticmethod
+    def languages(options) -> Set[str]:
+        """Return the languages supported by RapidOCR."""
+        # RapidOCR primarily supports Chinese, English and other languages
+        # but does not have the same language-specific models as Tesseract
+        # Return a set of common languages it works well with
+        return {'chi_sim', 'chi_tra', 'eng'}
+
+    @staticmethod
+    def get_orientation(input_file: Path, options) -> OrientationConfidence:
+        """Return the orientation detected by RapidOCR."""
+        # RapidOCR doesn't have built-in orientation detection like Tesseract
+        # Returning default value - could be enhanced with custom algo
+        return OrientationConfidence(angle=0, confidence=0.0)
+
+    @staticmethod
+    def get_deskew(input_file: Path, options) -> float:
+        """Return deskew angle (degrees)."""
+        # Return 0 as RapidOCR doesn't have direct deskew calculation
+        return 0.0
+
+    @staticmethod
+    def generate_hocr(
+        input_file: Path, output_hocr: Path, output_text: Path, options
+    ) -> None:
+        """Generate HOCR file from the OCR results."""
+        if not RAPIDOCR_AVAILABLE:
+            error_msg = f"rapidocr_onnxruntime is not installed: {IMPORT_ERROR_MESSAGE}"
+            log.error(error_msg)
+            raise ImportError(
+                f"{error_msg}\nInstall it with: pip install rapidocr_onnxruntime"
+            )
+
+        try:
+            # Load image
+            image = Image.open(input_file)
+            # Convert to numpy array for RapidOCR
+            img_array = np.array(image)
+
+            # Pre-process the image to improve detection
+            # Increase contrast
+            enhancer = ImageEnhance.Contrast(image)
+            image = enhancer.enhance(1.5)  # Increase contrast by 50%
+
+            # Or try padding the image
+            padded = Image.new(
+                image.mode, (image.width, image.height + 20), (255, 255, 255)
+            )
+            padded.paste(image, (0, 0))
+            img_array = np.array(padded)
```

Comment:
Use `with Image.new() as padded:` to ensure disposal of intermediate image after creating img_array.